app/controllers/submissions_controller.rb

```diff
@@ -168,21 +170,32 @@ def destroy
   end
 
   # POST
-  def mod_edit
-    if params.has_key? :soft_deleted
-      @submission.soft_deleted = (params[:soft_deleted] == "true")
-      if @submission.soft_deleted
-        @submission.soft_deleted_by = current_user.id
+  def mod_action
+    if params.has_key?(:reason) && params[:reason].present?
+      if params.has_key? :toggle_soft_delete
+        @submission.soft_deleted = !@submission.soft_deleted
+        if @submission.soft_deleted
+          @submission.soft_deleted_by = current_user.id
+        end
+        ModeratorLog.create(user_id: current_user.id, 
+                            target: @submission,
+                            action: "#{current_user.username} has #{@submission.soft_deleted ? 'soft deleted' : 'reverted soft deletion on'} #{@submission.display_title} by #{@submission.user.username}.",
+                            reason: params[:reason])
+      elsif params.has_key? :toggle_approve
+        @submission.approved = !@submission.approved
+        ModeratorLog.create(user_id: current_user.id, 
+                            target: @submission,
+                            action: "#{current_user.username} has #{@submission.approved ? 'approved' : 'disapproved'} #{@submission.display_title} by #{@submission.user.username}.",
+                            reason: params[:reason])
+      elsif params.has_key? :change_nsfw
+        @submission.nsfw_level = params[:change_nsfw].to_i
+        ModeratorLog.create(user_id: current_user.id, 
+                            target: @submission,
+                            action: "#{current_user.username} has changed the content level of #{@submission.display_title} by #{@submission.user.username} to #{nsfw_string(@submission.nsfw_level)}.",
+                            reason: params[:reason])
       end
+      @submission.save
```

Comment:
This code seems very repetitive across controllers. How about factoring out something like `ModeratorLog.log_action(@submission, :content_level, params[:reason])`, or even the entire block into something like `moderator_action(@target, %i[soft_delete approve change_nsfw]` where the array represents the types of actions which may be taken on the target?